lib/api/objectCopy.js

```diff
@@ -290,6 +291,11 @@ function objectCopy(authInfo, request, sourceBucket,
                         log.debug('no source object', { sourceObject });
                         return next(err, null, destBucketMD);
                     }
+                    // check if object data is in a cold storage
+                    const coldErr = verifyColdObjectAvailable(sourceObjMD);
+                    if (coldErr) {
```

Comment:
Nit: is it necessary/good practice to create a variable? Ignore if so
```suggestion
                    if (verifyColdObjectAvailable(sourceObjMD)) {
```

Reply:
same applies for some areas below

Reply:
the function returns an error so we have to store its value somewhere to return it